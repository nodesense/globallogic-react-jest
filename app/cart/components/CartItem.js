import React, {Component} from "react";

import {View, Text, TextInput, Button, StyleSheet} from "react-native";

export default class CartItem extends Component {
    //creation: stage 1
    //called once
    constructor(props) {
        super(props);
        console.log("cart item created", props.item.id);

        //update qty on creation cycle
        this.state = {
            qty: props.item.qty
        }
    }

    componentWillReceiveProps(nextProps) {
        //update qty on update cycle
        this.setState({
            qty: nextProps.item.qty
        });
    }

    //creation : stage 2
    //View is not ready
    //called once
    componentWillMount() {

    }

    handleChange(ctrl) {
        console.log("**HELLO**")
    }

    //render a view: stage 3
    //return a view
    //called multiple times
    //called once during creation
    render() {
        let {item} = this.props;
        
        console.log("CartItem render ", item.id);

        return (
            <View style={{ backgroundColor: 'powderblue'}}>
                <Text> {item.name} </Text>
                <TextInput style={[styles.inputContainer, styles.input]} 
                          value={this.state.qty.toString()}
                          onChangeText={ (text) => this.setState({
                              qty: parseInt(text) 
                          }) }
                          >
                </TextInput>

                <Text id="qty" > Qty: {item.qty}</Text>
                <Text > Price: {item.price}</Text>

                <Text > Total: {item.price * item.qty}</Text>

                <Button title="Update" 
                        onPress={() => this.props.onUpdate(item.id, this.state.qty)} >
                </Button>

                <Button title="Remove" 
                        onPress={ () => this.props.onRemove(item.id)}>
                </Button>
            </View>
        )
    }

    //creation: stage 4
    //called once
    //called after view ready
    componentDidMount() {

    }

    //destruction stage: 1
    //called once
    componentWillUnmount() {
        console.log("cart item unmount ", this.props.item.id);
    }
}

const styles = StyleSheet.create({
    inputContainer: {
        borderLeftWidth: 4,
        borderRightWidth: 4,
        height: 70
      },
      input: {
        height: 70,
        backgroundColor: '#ffffff',
        paddingLeft: 15,
        paddingRight: 15
      }
});