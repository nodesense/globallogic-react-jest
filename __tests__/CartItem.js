
import 'react-native';
import React from 'react';
import CartItem from '../app/cart/components/CartItem';

import {shallow, mount} from "enzyme";

import {Text, Button} from "react-native";

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

xit('CartItem Snapshot', () => {
  let item =  {id: 1, name: 'Product 1', price: 100, qty: 10};
  const tree = renderer.create(
    <CartItem  item={item} />
  );

  const treeJson = tree.toJSON();


  console.log("JSON ", treeJson);
  expect(treeJson).toMatchSnapshot();

  const testInstance = tree.root;

  testInstance.instance.handleChange(null);
  expect(testInstance.findAllByType(Button)[0].props.title).toBe('Update');

  
  // expect(testInstance.findByProps({id: "qty"}).children).toEqual('Qty: 10');

  expect(testInstance.findAllByType(Text)[1].children).toBe(' Qty: ');



});

xit("CartItem Test",  ()=> {
    let item =  {id: 1, name: 'Product 1', price: 100, qty: 10};

    let wrapper = shallow(<CartItem item={item} />);
    wrapper.setProps({ item: item  });
    
  
    let instance = wrapper.instance();
     
    //wrapper.find('Button[title="+1"]').simulate('click');
    //wrapper.find('Button').simulate('click');
  
    //await wrapper.update();
  
    // console.log("**COUNTER ", wrapper.state("counter"));
  
    // console.log("***TEXT", wrapper.text());
    // expect(wrapper.state("counter")).toBe(1003);
    // //use at, last, first
    // expect(wrapper.find("Text").at(1).text()).toContain("1003");
  
    //console.log("FULL HTML", wrapper.html());
    
  });
  