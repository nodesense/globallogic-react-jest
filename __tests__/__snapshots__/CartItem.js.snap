// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
     
    Product 1
     
  </react-native-mock>
  <react-native-mock />
  <react-native-mock>
     Qty: 
    10
  </react-native-mock>
  <react-native-mock>
     Price: 
    100
  </react-native-mock>
  <react-native-mock>
     Total: 
    1000
  </react-native-mock>
  <react-native-mock />
  <react-native-mock />
</react-native-mock>
`;
